src/secure-postgresql-db/index.ts

```diff
@@ -0,0 +1,551 @@
+        }),
+      }),
+    );
+
+    this.alarms.push(
+      new Alarm(this, 'DbAlarmStorageSpace', {
+        comparisonOperator: ComparisonOperator.LESS_THAN_THRESHOLD,
+        threshold: this._props.maxAllocatedStorage! * (this._props.alarmProps?.freeStorageThresholdPct! / 100),
+        evaluationPeriods: 2,
+        metric: this.instance.metricFreeStorageSpace({
+          statistic: 'avg',
+          period: Duration.minutes(15),
+        }),
+      }),
+    );
+
+    this.alarms.push(
+      new Alarm(this, 'DbAlarmOpenConnections', {
+        comparisonOperator: ComparisonOperator.GREATER_THAN_THRESHOLD,
+        threshold: this._props.alarmProps?.maxOpenConnections!,
+        evaluationPeriods: 2,
+        metric: this.instance.metricDatabaseConnections({
+          statistic: 'avg',
+          period: Duration.minutes(15),
+        }),
+      }),
+    );
+
+    if (this._alarmTopic) {
+      this._registerAlarms(this._alarmTopic);
+    }
+  }
+
+  private _createBackupPlan() {
+    this.backupPlan = new BackupPlan(this, 'DbBackupPlan');
+
+    this.backupPlan.addRule(new BackupPlanRule(this._props.backupPlanRuleProps!));
+
+    this.backupPlan.addSelection('Selection', {
+      resources: [BackupResource.fromRdsDatabaseInstance(this.instance)],
+    });
+  }
+
+  private _createDbInstance() {
+    this.instance = new DatabaseInstance(this, 'DbInstance', {
+      instanceIdentifier: this._props.instanceIdentifier ?? undefined,
+      instanceType: this._props.instanceType,
+      engine: DatabaseInstanceEngine.postgres({
+        version: this._props.postgresEngineVersion!,
+      }),
+      multiAz: this._props.multiAz,
+      port: this._props.port,
+      vpc: this._vpc,
+      vpcSubnets: { subnetType: SubnetType.PRIVATE_WITH_EGRESS },
+      securityGroups: [
+        this.securityGroup as ISecurityGroup,
+      ].concat(this._props.additionalSecurityGroups!),
+      allocatedStorage: this._props.allocatedStorage,
+      maxAllocatedStorage: this._props.maxAllocatedStorage,
+      cloudwatchLogsExports: ['postgresql', 'upgrade'],
+      cloudwatchLogsRetention: this._props.cloudWatchLogRetention,
+      credentials: Credentials.fromSecret(this.credentialSecret),
+      parameterGroup: createRDSLogEnabledParamGroup(this, {
+        postgresEngineVersion: this._props.postgresEngineVersion!,
+      }),
+      removalPolicy: RemovalPolicy.RETAIN,
+      storageEncrypted: true,
+      storageEncryptionKey: this.encryptionKey,
+      backupRetention: Duration.days(7),
+      ...this._props.overrideInstanceProps,
+    });
+
+    if (this._props.snapshotArn) {
+      const cfnDbInstance = this.instance.node.defaultChild as CfnDBInstance;
+      cfnDbInstance.addPropertyOverride('DbSnapshotIdentifier', this._props.snapshotArn);
+    }
+  }
+
+  private _createEncryptionKey() {
+    this.encryptionKey = new Key(this, 'DbKmsKey', {
+      description: `CDK generated database encryption key. STACK - ${cdk.Stack.of(this)}`,
```

Comment:
May be we should spell out that, "Key is utilzied for encrypting database identified by {instanceIdentifier}"

Reply:
Added description, will include instance identifier if user specified. Unfortunately can't add any auto-generated instance id to the description, since it would make a cyclical dependency between the key and db instance